packages/core/src/app/ui/form/FormFieldError.tsx

```diff
@@ -10,30 +10,44 @@ export interface FormFieldErrorProps {
 }
 
 const FormFieldError: FunctionComponent<FormFieldErrorProps> = ({ name, testId, errorId }) => {
+    // Get form context to check for errors
+    const formikContext = useFormikContext<{ [key: string]: any }>();
+    const hasError = formikContext?.errors?.[name] && formikContext?.touched?.[name];
+    
     const renderMessage = useCallback(
         (message: string) => (
-            <ul className="form-field-errors" data-test={testId}>
-                <li className="form-field-error">
-                    <label
-                        aria-live="polite"
-                        className="form-inlineMessage"
-                        htmlFor={name}
-                        id={errorId}
-                        role="alert"
-                    >
-                        {message}
-                    </label>
-                </li>
-            </ul>
+            <label
+                aria-live="polite"
+                className="form-inlineMessage"
+                htmlFor={name}
+                id={errorId}
+                role="alert"
+            >
+                {message}
+            </label>
         ),
-        [errorId, name, testId],
+        [errorId, name],
     );
 
     return (
         <FormContext.Consumer>
-            {({ isSubmitted }) =>
-                isSubmitted && <ErrorMessage name={name} render={renderMessage} />
-            }
+            {({ isSubmitted }) => (
+                <ul className="form-field-errors" data-test={testId}>
+                    <li className="form-field-error">
+                        {/* Show error message when form is submitted and there's an error */}
+                        {isSubmitted && <ErrorMessage name={name} render={renderMessage} />}
+                        
+                        {/* When there's no error or form isn't submitted, render a hidden element with the same ID */}
+                        {(!isSubmitted || !hasError) && (
+                            <span 
+                                aria-hidden="true"
+                                className="is-srOnly" 
+                                id={errorId} 
+                            />
+                        )}
```

Comment:
In my manual testing, I saw two elements with the same id.


I'm working on a solution for the raised issue.

cc @animesh1987 

```
<li class="form-field-error">
    <label 
        aria-live="polite" 
        class="form-inlineMessage" 
        for="shippingAddress.lastName" 
        id="lastNameInput-field-error-message" 
        role="alert">
        Last Name is required
    </label>
    <span 
        aria-hidden="true" 
        class="is-srOnly" 
        id="lastNameInput-field-error-message">
    </span>
</li>
```
<img width="2559" alt="image" src="https://github.com/user-attachments/assets/5c3ef63e-1c1e-4917-b28f-199a07a73f22" />